### PR TITLE
statistics: uniform the calculation of pseudo stats.

### PR DESCRIPTION
**Pull request by winoros**
To make `Selectivity` not a method of `statistics.Table`. It's inner logic should not rely on `Table.Pseudo`.

### CODE REVIEW

File: plan/stats.go [hunk 1]
```diff
@@ -19,13 +19,15 @@ import (
...
 	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/statistics"
 	log "github.com/sirupsen/logrus"
 )
 
 // statsInfo stores the basic information of statistics for the plan's output. It is used for cost estimation.
 type statsInfo struct {
 	count       float64
 	cardinality []float64
+	colHist     []statistics.Column
```
Comment (lamxTyler) [inline plan/stats.go]: Seems that it is not used?

Comment (winoros) [reply plan/stats.go]: Oh it's something should be done in next pr.

**Comment by jackysp** [comment]
CI failed @winoros .

**Pull request merged**